haystack/nodes/prompt/prompt_template.py

```diff
@@ -218,8 +325,18 @@ def __init__(self, prompt: str, output_parser: Optional[Union[BaseOutputParser,
         name, prompt_text = "", ""
 
         try:
+            if prompt in LEGACY_DEFAULT_TEMPLATES:
+                warnings.warn(
+                    f"You're using a legacy prompt template '{prompt}', "
+                    "we strongly suggest you use prompts from the official Haystack PromptHub: "
+                    "https://prompthub.deepset.ai/"
+                )
+                name = prompt
+                prompt_text = LEGACY_DEFAULT_TEMPLATES[prompt]["prompt"]
```

Comment:
I know this is a bit of a nitpick here, but why not use PromptTemplate instances as values in our legacy dict? Then these two lines of code below would be simpler, I think, and perhaps "less scary" that this double hop into dict. Just a thought.

Reply:
I tried that, but unfortunately the two depend too strictly one from the other and it creates an unsolvable circular dependency. This seems to be the most straightforward way to go.